cmd/controller/main.go

```diff
@@ -30,14 +30,21 @@ func main() {
 	}
 
 	metadataClient := metadata.NewClient(metadata.WithInstrumentation(m.Registry()))
+	var location string
 
-	server, err := app.GetServer(logger, hcloudClient, metadataClient)
-	if err != nil {
-		level.Error(logger).Log(
-			"msg", "failed to fetch server",
-			"err", err,
-		)
-		os.Exit(1)
+	if s := os.Getenv("HCLOUD_SERVER_OVERRIDE_LOCATION_NAME"); s != "" {
+		location = s
+	} else {
+		server, err := app.GetServer(logger, hcloudClient, metadataClient)
```

Comment:
`metadataClient` is only used in this branch. Can you move the `metadataClient := metadata.NewClient()` line here?